src/mumble/Database.cpp

```diff
@@ -55,42 +54,57 @@ Database::Database(const QString &dbname) {
 	datapaths << QDir::currentPath();
 	datapaths << qApp->applicationDirPath();
 	datapaths << qs.value(QLatin1String("InstPath")).toString();
-	bool found = false;
-
-	for (i = 0; (i < datapaths.size()) && ! found; i++) {
-		if (!datapaths[i].isEmpty()) {
-			// Try the legacy path first, and use it if it exists.
-			// If it doesn't, use the new, non-hidden version.
-			QFile legacyDatabaseFile(datapaths[i] + QLatin1String("/.mumble.sqlite"));
-			if (legacyDatabaseFile.exists()) {
-				db.setDatabaseName(legacyDatabaseFile.fileName());
-				found = db.open();
+	datapaths.removeAll(QLatin1String(""));
+	datapaths.removeDuplicates();
+
+	// Try to find an existing database
+	foreach(const QString &datapath, datapaths) {
+		// Try the legacy path first, and use it if it exists.
+		// If it doesn't, use the new, non-hidden version.
+		QFile legacyDatabaseFile(datapath + QLatin1String("/.mumble.sqlite"));
+		if (legacyDatabaseFile.exists()) {
+			db.setDatabaseName(legacyDatabaseFile.fileName());
+			if (db.open()) {
+				return true;
 			}
-			if (found) {
-				break;
-			}
-			QFile databaseFile(datapaths[i] + QLatin1String("/mumble.sqlite"));
-			if (databaseFile.exists()) {
-				db.setDatabaseName(databaseFile.fileName());
-				found = db.open();
+		}
+		QFile databaseFile(datapath + QLatin1String("/mumble.sqlite"));
+		if (databaseFile.exists()) {
+			db.setDatabaseName(databaseFile.fileName());
+			if (db.open()) {
+				return true;
 			}
 		}
 	}
 
-	if (! found) {
-		for (i = 0; (i < datapaths.size()) && ! found; i++) {
-			if (!datapaths[i].isEmpty()) {
-				QDir::root().mkpath(datapaths[i]);
-				QFile f(datapaths[i] + QLatin1String("/mumble.sqlite"));
-				db.setDatabaseName(f.fileName());
-				found = db.open();
-			}
+	// There is no existing database, so we create one
+	foreach(const QString &datapath, datapaths) {
+		QDir::root().mkpath(datapath);
+		QFile f(datapath + QLatin1String("/mumble.sqlite"));
+		db.setDatabaseName(f.fileName());
+		if (db.open()) {
+			return true;
 		}
 	}
+	return false;
+}
 
-	if (! found) {
-		QMessageBox::critical(nullptr, QLatin1String("Mumble"), tr("Mumble failed to initialize a database in any\nof the possible locations."), QMessageBox::Ok | QMessageBox::Default, QMessageBox::NoButton);
-		qFatal("Database: Failed initialization");
+Database::Database(const QString &dbname) {
+	db = QSqlDatabase::addDatabase(QLatin1String("QSQLITE"), dbname);
+	if (!g.s.qsDatabasePath.isEmpty()) {
+		QFile configuredLocation(g.s.qsDatabasePath);
+		if (configuredLocation.exists()) {
```

Comment:
If it doesn't exist we should output an error (and maybe even quit Mumble)

Reply:
I think it would be good to show a warning if the configured path isn't found. Maybe it would be helpful to add an option to create a new DB to that dialog.

Reply:
Sounds good